federation/pkg/federation-controller/ingress/ingress_controller.go

```diff
@@ -707,29 +703,43 @@ func (ic *IngressController) reconcileIngress(ingress types.NamespacedName) {
 				glog.V(4).Infof(logStr, "Not transferring")
 			}
 			// Update existing cluster ingress, if needed.
-			if util.ObjectMetaEquivalent(desiredIngress.ObjectMeta, clusterIngress.ObjectMeta) &&
-				reflect.DeepEqual(desiredIngress.Spec, clusterIngress.Spec) &&
-				reflect.DeepEqual(baseIngress.Status.LoadBalancer.Ingress, clusterIngress.Status.LoadBalancer.Ingress) {
+			if util.ObjectMetaEquivalent(baseIngress.ObjectMeta, clusterIngress.ObjectMeta) &&
+				reflect.DeepEqual(baseIngress.Spec, clusterIngress.Spec) {
 				glog.V(4).Infof("Ingress %q in cluster %q does not need an update: cluster ingress is equivalent to federated ingress", ingress, cluster.Name)
 			} else {
 				glog.V(4).Infof("Ingress %s in cluster %s needs an update: cluster ingress %v is not equivalent to federated ingress %v", ingress, cluster.Name, clusterIngress, desiredIngress)
-				if !util.ObjectMetaEquivalent(desiredIngress.ObjectMeta, clusterIngress.ObjectMeta) {
-					// Merge any annotations on the federated ingress onto the underlying cluster ingress,
-					// overwriting duplicates.
-					for key, val := range baseIngress.ObjectMeta.Annotations {
-						desiredIngress.ObjectMeta.Annotations[key] = val
-					}
-					ic.eventRecorder.Eventf(baseIngress, api.EventTypeNormal, "UpdateInCluster",
-						"Updating ingress in cluster %s", cluster.Name)
-
-					operations = append(operations, util.FederatedOperation{
-						Type:        util.OperationTypeUpdate,
-						Obj:         desiredIngress,
-						ClusterName: cluster.Name,
-					})
-					// TODO: Transfer any readonly (target-proxy, url-map etc) annotations from the master cluster to the federation, if this is the master cluster.
-					// This is only for consistency, so that the federation ingress metadata matches the underlying clusters.  It's not actually required.
+				objMeta, err := conversion.NewCloner().DeepCopy(clusterIngress.ObjectMeta)
```

Comment:
optional: I liked the previous code where we did all of this only if ObjectMeta is different.
Dont need to do all of this (recompute ObjectMeta) if ObjectMetaEquivalent is true.